app/client/templates/elements/eventsTable.js

```diff
@@ -49,8 +49,9 @@ Template['elements_event_table'].helpers({
             searchQuery = TemplateVar.get('search'),
             limit = TemplateVar.get('limit'),
             collection = Events,
-            selector = this.ids ? {_id: {$in: this.ids}} : {};
-
+            selector = this.ids ? {_id: {$in: this.ids.slice(Number((limit+1)*-1))}} : {}; 
```

Comment:
I understand what you where doing. But this has the cost of a wrong selection if the IDs are to sorted.

See this example:
```js


MyColl.find({id: {$in: [1, 2, 3]}, {sort: {_id: 1}, limit: 2}})
// -> will give me 1, 2

// If you pass in `[5, 3, 7, 2, 1]`, but cut it to items 2 before
MyColl.find({id: {$in: [5, 3]}, {sort: {_id: 1}, limit: 2}})
// -> gives you 5,3; as the cursor could never sort it before properly, while

MyColl.find({id: {$in: [5, 3, 7, 2, 1]}, {sort: {_id: 1}, limit: 2}})
// -> would give you 1, 2
```

So yes, your way is faster, but its slower my way because it *sorts* the results before. And if you have a table where you want to sort by multiple properties, thats important.

Best check if the IDs we are having in `this.ids` are time sorted and it should be fine.